api/v1alpha1/dnspolicy_types.go

```diff
@@ -178,6 +180,21 @@ func (p *DNSPolicy) GetStatus() kuadrantgatewayapi.PolicyStatus {
 
 func (p *DNSPolicy) Kind() string { return p.TypeMeta.Kind }
 
+func (p *DNSPolicy) List(ctx context.Context, c client.Client, namespace string) []kuadrantgatewayapi.Policy {
```

Comment:
why not fuction instead of method? 

```go
func DNSPolicyList(ctx context.Context, c client.Client, namespace string) []kuadrantgatewayapi.Policy {
}
```

Reply:
Because of where it is being used. The `List` method was added to the [Policy Interface](https://github.com/Boomatang/kuadrant-operator/blob/ca84e63edac522a0c28ac727f03946f0d155fafa/pkg/library/gatewayapi/types.go#L25) as it required in the mapping functions.

Reply:
The bad smell is that you are not using the receiver, thus looks like a dog, walks like a dog and barks like a dog, but it is a cat. (dog => function, cat => method)

Reply:
One of the previous iterations of this PR did not have this. Instead it listed the policies from the cluster based on the type passed in. There was a switch within the mapper function. It was suggested that this was the cleaner approach. I understand why you don't like it.

Reply:
The method could have been name sth like `ListAllOfKind`.